### PR TITLE
Leonid: readme updates; minor effort refactoring

### PR DESCRIPTION
**Pull request by Leonid-98**
* Added documentation
* Changed effort to duty cycle

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,20 +1,59 @@
 # Robotont mbed mainboard firmware
 
-Simplistic firmware for the Robotex platform mainboard.
+Firmware for the Robotont platform mainboard.
+Main tasks:
+ * Movement control (Three PWM motors)
+ * Addressable LED control
+ * OLED screen control
+ * Communication with battery monitor board (TO BE ADDED)
```
Comment (Veix123) [inline README.md]: let's rename to power management board, to be consistent with other documentation

File: README.md [hunk 2]
```diff
@@ -1,20 +1,59 @@
...
+|Type|Argument|Description|Format|
+|---|---|---|---|
+|Output|`ODOM`|Odometry data(*)|`"ODOM:{pos_x}:{pos_y}:{pos_z}:{speed_x}:{speed_y}:{speed_z}\r\n"`|
+|Input|`RS`|Robot's x, y and rotation speed(*)|`"RS:{speed_x}:{speed_y}:{speed_z}\r\n"`|
+|Input|`MS`|Individual motor speed(*)|`"MS:{speed_0}:{speed_1}:{speed_2}\r\n"`|
+|Input|`DC`|Individual motor duty cycle(**)|`"DC:{motor_0}:{motor_1}:{motor_2}\r\n"`|
+|Input|`OR`|Resets odometry data|`"OR\r\n"`|
 
-In the same format You can also send data to the mainboard to set the motor speeds (from -100 to 100).
+(\*) *All speed units are arbitrary in range (-100:100).* </br>
```
Comment (Veix123) [inline README.md:24]: Needs attention. For 'RS' and 'MS' the speed should be in standard units of m/s. For 'RS' it has to be in the Robot's coordinates and for 'MS' it should be the linear speed of the wheel along the axis perpendicular to the motor shaft.
We should clearly note that in 'DC' mode, velocity controllers are disabled and the values are directly set as PWM signal duty cycles for each motor.

File: README.md [hunk 3]
```diff
@@ -1,20 +1,59 @@
...
+(\*\*) *Robot will enter "manual mode", meaning that motors will rotate with given duty cycle until `MS` or `RS` commands is received. Duty cycle range is (-100:100), where minus mark means negative rotation direction. Threshold to stop is 10%.*
+
+## Updating
```
Comment (Veix123) [inline README.md]: I would use 'uploading' here.

File: docs/.images/fw_arch_if.png
Comment (Veix123) [inline docs/.images/fw_arch_if.png:1]: "middelware" -> "middleware"

File: docs/.images/fw_architecture.png
Comment (Veix123) [inline docs/.images/fw_architecture.png:1]: "middelware" -> "middleware"

File: docs/firmware_design.md [hunk 4]
```diff
@@ -0,0 +1,132 @@
...
+
+## Interface layer
+
+<img align="top" src=".images/fw_arch_if.png" width="800" />
+
+</br>
+
+The purpose of the interface layer is to isolate `STM32F4 HAL` functions from Robotont's application, which are used for communication with MCU peripherals.
+
+This approach provides a significant advantage in terms of portability: If in the future `STM32F4 HAL` is no longer needed (either because the MCU family changed or the HAL framework), developers do not need to edit files in the upper layer. It is sufficient to change only the interface layer files.
```
Comment (Veix123) [inline docs/firmware_design.md]: This approach provides a significant advantage in terms of portability. If the MCU family and HAL framework undergo changes, developers need only modify the interface layer files, without editing upper layer files.

File: src/app/main.c [hunk 5]
```diff
@@ -1,118 +1,109 @@
...
+#include "odom.h"
+#include "peripheral.h"
+#include "pid.h"
+#include "ssd1306.h"
+#include "ssd1306_tests.h"
+#include "system_hal.h"
+#include "timerif.h"
+#include "usbif.h"
+
+#define MAX_LIN_VEL 0.4 // m/s
```
Comment (Veix123) [inline src/app/main.c]: should these be moved to main.h as well?

**Review by Veix123 — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (Leonid-98) [reply README.md]: Renamed

File: README.md (hunk 3 above)
Comment (Leonid-98) [reply README.md]: Renamed

File: docs/.images/fw_architecture.png
Comment (Leonid-98) [reply docs/.images/fw_architecture.png:1]: Fixed

File: src/app/main.c (hunk 5 above)
Comment (Leonid-98) [reply src/app/main.c]: Actually, they already defined in movement.h, so I just remove them from main file

File: docs/firmware_design.md (hunk 4 above)
Comment (Leonid-98) [reply docs/firmware_design.md]: Changed to this text

File: docs/.images/fw_arch_if.png
Comment (Leonid-98) [reply docs/.images/fw_arch_if.png:1]: Changed

File: README.md (hunk 2 above)
Comment (Leonid-98) [reply README.md:24]: PID will have to be re-tuned.
I will figure this out, but it takes a bit of time.

**Review by Veix123 — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (Veix123) [reply README.md:24]: moved this to a separate issue to accept other changes now #19

**Pull request merged**